pkg/wallet/wallet.go

```diff
@@ -151,7 +163,12 @@ func (w *Wallet) Unprotect(password string) *WalletError {
 		return &WalletError{fmt.Errorf("opening the private key seal: %w", err), utils.WrongPassword}
 	}
 
-	w.KeyPair.PrivateKey = pk
+	privateKey, err := CheckPrivKeyVersion(pk)
+	if err != nil {
+		return &WalletError{fmt.Errorf("while checking private key version: %w", err), utils.ErrInvalidPrivateKey}
```

Comment:
What is this horror ? Why do we need a Wallet Error struct ?

Reply:
This function returns a WalletError to be able to specify an error and a error code. The error code is included in the response http call or to the event send to wails. This way we can return an error and the type of the error with the error code. The error code can be used to translate the error message shown to the user.

Reply:
:see_no_evil: